### PR TITLE
13172 - Undo button can now be assigned a hotkey by module designer

### PR DESCRIPTION
**Pull request by Cattlesquat**
Still left it no key by default, but now it can at least be assigned one.

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java [hunk 1]
```diff
@@ -147,6 +150,22 @@ public void propertyChange(PropertyChangeEvent evt) {
       }
     });
     stepIconConfig.fireUpdate();
+    
+    final NamedHotKeyConfigurer stepKeyConfig = new NamedHotKeyConfigurer("stepHotKey", Resources.getString("BasicLogger.step_forward_hotkey"), stepKeyListener.getNamedKeyStroke());  //$NON-NLS-1$ //$NON-NLS-2$
+    GlobalOptions.getInstance().addOption(stepKeyConfig);
+    stepKeyConfig.addPropertyChangeListener(new PropertyChangeListener() {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java:156]: Could make this a lambda.

File: vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java [hunk 2]
```diff
@@ -159,16 +178,21 @@ public void propertyChange(PropertyChangeEvent evt) {
     });
     undoIconConfig.fireUpdate();
 
-    final NamedHotKeyConfigurer stepKeyConfig = new NamedHotKeyConfigurer("stepHotKey", Resources.getString("BasicLogger.step_forward_hotkey"), stepKeyListener.getNamedKeyStroke());  //$NON-NLS-1$ //$NON-NLS-2$
-    GlobalOptions.getInstance().addOption(stepKeyConfig);
-    stepKeyConfig.addPropertyChangeListener(new PropertyChangeListener() {
+    final NamedHotKeyConfigurer undoKeyConfig = new NamedHotKeyConfigurer("undoHotKey", Resources.getString("BasicLogger.undo_hotkey"), undoKeyListener.getNamedKeyStroke()); //$NON-NLS-1$ //$NON-NLS-2$
+    GlobalOptions.getInstance().addOption(undoKeyConfig);
+    undoKeyConfig.addPropertyChangeListener(new PropertyChangeListener() {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java:183]: And this one.

**Comment by uckelman** [comment]
This is fine.

That said, we really screwed up way back at the start by not assigning the standard system hotkeys to standard things like Undo and Redo. That ship has sailed for V3---we can't now be sure that the standard hotkeys don't conflict with what's in some module----but it's a thing I intend to do right in V4.

File: vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java (hunk 2 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/BasicLogger.java]: Lambdas up in latest push.

**Comment by Cattlesquat** [comment]
Agree on "claiming" some of the core default standard keys for Vassal. Although I also think that V4 should, like an IDE, have a near-religious commitment to allowing the module designer to redefine pretty much any key. If some jackwagon wants Ctrl+Z to mean "Fire Forward Phaser Banks" then we should let him.

**Comment by uckelman** [comment]
Agreed, the capability to rebind keys is a good thing, so are unsurprising defaults.

**Pull request merged**